rust/src/smb/smb.rs

```diff
@@ -2189,3 +2190,143 @@ pub extern "C" fn rs_smb_state_get_event_info(event_name: *const std::os::raw::c
+    let default_port = CString::new("445").unwrap();
+    let mut stream_depth = SMB_CONFIG_DEFAULT_STREAM_DEPTH;
+    let parser = RustParser {
+        name: PARSER_NAME.as_ptr() as *const std::os::raw::c_char,
+        default_port: default_port.as_ptr(),
+        ipproto: IPPROTO_TCP,
+        probe_ts: None,
+        probe_tc: None,
+        min_depth: 0,
+        max_depth: 16,
+        state_new: rs_smb_state_new,
+        state_free: rs_smb_state_free,
+        tx_free: rs_smb_state_tx_free,
+        parse_ts: rs_smb_parse_request_tcp,
+        parse_tc: rs_smb_parse_response_tcp,
+        get_tx_count: rs_smb_state_get_tx_count,
+        get_tx: rs_smb_state_get_tx,
+        tx_comp_st_ts: 1,
+        tx_comp_st_tc: 1,
+        tx_get_progress: rs_smb_tx_get_alstate_progress,
+        get_de_state: rs_smb_state_get_tx_detect_state,
+        set_de_state: rs_smb_state_set_tx_detect_state,
+        get_events: Some(rs_smb_state_get_events),
+        get_eventinfo: Some(rs_smb_state_get_event_info),
+        get_eventinfo_byid : Some(rs_smb_state_get_event_info_by_id),
+        localstorage_new: None,
+        localstorage_free: None,
+        get_files: Some(rs_smb_getfiles),
+        get_tx_iterator: Some(rs_smb_state_get_tx_iterator),
+        get_tx_data: rs_smb_get_tx_data,
+        apply_tx_config: None,
+        flags: APP_LAYER_PARSER_OPT_ACCEPT_GAPS,
+        truncate: Some(rs_smb_state_truncate),
+    };
+
+    let ip_proto_str = CString::new("tcp").unwrap();
+
+    if AppLayerProtoDetectConfProtoDetectionEnabled(
+        ip_proto_str.as_ptr(),
+        parser.name,
+    ) != 0
+    {
+        let alproto = AppLayerRegisterProtocolDetection(&parser, 1);
+        ALPROTO_SMB = alproto;
+        if register_pattern_probe() < 0 {
+            return;
+        }
+
+        let have_cfg = AppLayerProtoDetectPPParseConfPorts(ip_proto_str.as_ptr(),
+                    IPPROTO_TCP as u8, parser.name, ALPROTO_SMB, 0,
+                    MIN_REC_SIZE, rs_smb_probe_tcp, rs_smb_probe_tcp);
+
+        if have_cfg == 0 {
+            AppLayerProtoDetectPPRegister(IPPROTO_TCP as u8, parser.default_port, ALPROTO_SMB,
+                                          0, MIN_REC_SIZE, STREAM_TOSERVER, rs_smb_probe_tcp, rs_smb_probe_tcp);
+        }
+
+        if AppLayerParserConfParserEnabled(
+            ip_proto_str.as_ptr(),
+            parser.name,
+        ) != 0
+        {
+            let _ = AppLayerRegisterParser(&parser, alproto);
+        }
+        SCLogDebug!("Rust SMB parser registered.");
+        let retval = conf_get("app-layer.protocols.smb.stream-depth");
+        if let Some(val) = retval {
+            let val = val.parse::<i32>().unwrap();
+            if val < 0 {
+                SCLogError!("invalid value for stream-depth");
+            } else {
+                stream_depth = val as u32;
+           }
+            AppLayerParserSetStreamDepth(IPPROTO_TCP as u8, ALPROTO_SMB, stream_depth);
```

Comment:
@inashivb This is now called only if we have `app-layer.protocols.smb.stream-depth` in the config, which was not the case before 

Suggested fix is 
```
diff --git a/rust/src/smb/smb.rs b/rust/src/smb/smb.rs
index 054e65ab8..482ae8a78 100644
--- a/rust/src/smb/smb.rs
+++ b/rust/src/smb/smb.rs
@@ -2288,8 +2288,8 @@ pub unsafe extern "C" fn rs_smb_register_parser() {
             } else {
                 stream_depth = val as u32;
            }
-            AppLayerParserSetStreamDepth(IPPROTO_TCP as u8, ALPROTO_SMB, stream_depth);
         }
+        AppLayerParserSetStreamDepth(IPPROTO_TCP as u8, ALPROTO_SMB, stream_depth);
     } else {
         SCLogDebug!("Protocol detector and parser disabled for SMB.");
     }
```

Reply:
Waiting on [this](https://github.com/OISF/suricata/pull/6364) to get merged/approved then do this fix. Thanks, @catenacyber !

Reply:
Ok, letting you handle this...